plugin/completion.py

```diff
@@ -41,10 +42,74 @@
 
 
 class LspResolveDocsCommand(sublime_plugin.TextCommand):
+    def run(self, edit: sublime.Edit, item: dict) -> None:
+        detail = item.get('detail') or ""
+        documentation = minihtml(item.get("documentation"), self.view)
+
+        # don't show the detail in the cooperate AC popup,
+        # if it is alread shown in the AC details filed.
+        self.is_detail_shown = bool(detail)
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+        if not detail or not documentation:
+            # To make sure that the details or documentation fields don't exist
+            # we resove the completion item.
+            # If those fields appear after the item is resolved
+            # update the popup.
+            self.do_resolve(item)
+
+    def show_popup(self, content: str) -> None:
+        mdpopups.show_popup(
+            self.view,
+            content,
+            flags=sublime.COOPERATE_WITH_AUTO_COMPLETE,
+            max_width=480,
+            max_height=333,
+            allow_code_wrap=True
+        )
+
+    def update_popup(self, content: str) -> None:
+        mdpopups.update_popup(
+            self.view,
+            content
+        )
+
+    def do_resolve(self, item: dict) -> None:
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].begin())
+        if not session:
+            return
+
+        client = client_from_session(session)
+        if not client:
+            return
+
+        completion_provider = session.get_capability('completionProvider')
+        has_resolve_provider = completion_provider and completion_provider.get('resolveProvider', False)
+        if has_resolve_provider:
+            client.send_request(Request.resolveCompletionItem(item), self.handle_resolve_response)
+
+    def handle_resolve_response(self, item: dict) -> None:
+        documentation = minihtml(item.get("documentation"), self.view)
+        detail = item.get('detail') or ""
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+    def get_content(self, documentation: str, detail: str) -> str:
+        content = ""
+        if detail and not self.is_detail_shown:
+            content += """<div class='highlight' style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(detail)
+
+        if documentation:
+            content += """<div style='margin: 10px'>
+                <p>{}</p>
```

Comment:
I don't know if minihtml handles HTML per spec but having other block elements inside `p` tag is not allowed and will make the parser move the contents outside of it.

Whether minihtml follows that part of the spec or not, I'd rather follow it and not have `p` element since you don't know what elements rendered content will include. Let's change it to div (with a custom class if needed to style it).

Reply:
Thanks :) for each case I will use a div